### ISSUE TITLE
Overwrite `:` in rex(...) to act as range

### ISSUE THREAD

**Issue by robertzk** [kevinushey/rex#30]
What do you think of this syntax?

```r
grepl(rex("a":"e"), c("b","d","f")) # TRUE TRUE FALSE
```

**Comment by jimhester** [comment]
Personally I think alphabetic ranges should be handled by using `one_of()` and subsetting `base::letters` (`base::LETTERS` for uppercase).

However because we also define letters to be `[[:alpha:]]+` if you try to do so it breaks.

```r
rex(one_of(letters[1:10]))
# [\[\[:alpha:\]\]+NANANANANANANANANA]
```
Fully qualifying the variable works however
```r
rex(one_of(base::letters[1:10]))
# [abcdefghij]
```

Due to this your proposal might have some merit.  I just hate to create an overload for one particular situation, although I guess it is fairly common.

**Comment by robertzk** [comment]
Naturally, `:` would only be overwritten *within* the evaluating context, rather than as a global export. Anyway, it's up to you, just a random idea I had.

**Comment by jimhester** [comment]
That is true, I don't think it is a bad idea and in light of the issue I raised it might be worth doing I will reopen this and we can let it percolate a bit before deciding.

**Comment by jimhester** [comment]
Implemented in ceef6ad4e739445fe7ddeb3d7b8acc8708caa55d
